### PR TITLE
Added all records page ui

### PR DESCRIPTION
**Pull request by vishwjeet-pawar03**
## Description

[Provide a brief description of the changes in this PR]

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes. Provide instructions so we can reproduce.]

### Test Configuration

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [ ] **Search capabilities** are working correctly
- [ ] **Knowledge search** is functioning properly
- [ ] **Connector indexing** is working as expected
- [ ] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
[Attach screenshots/videos showing the new behavior]

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

### CODE REVIEW

File: backend/python/app/connectors/services/base_arango_service.py [hunk 1]
```diff
@@ -2440,7 +2440,7 @@ async def reindex_single_record(self, record_id: str, user_id: str, org_id: str,
                     }
 
                 user_role = await self.get_user_kb_permission(kb_context["kb_id"], user_key)
-                if user_role not in ["OWNER", "WRITER", "READER"]:
+                if not user_role:
```
[the remark this answers is not in the document]
Comment (vishwjeet-pawar03) [reply backend/python/app/connectors/services/base_arango_service.py:2443]: as all permissions are passed so no need to maintain an array

File: backend/python/app/connectors/services/base_arango_service.py [hunk 2]
```diff
@@ -2455,7 +2455,7 @@ async def reindex_single_record(self, record_id: str, user_id: str, org_id: str,
                 permission_result = await self._check_record_permissions(record_id, user_key)
                 user_role = permission_result.get("permission")
 
-                if not user_role or user_role not in ["OWNER", "WRITER","READER"]:
+                if not user_role:
```
[the remark this answers is not in the document]
Comment (vishwjeet-pawar03) [reply backend/python/app/connectors/services/base_arango_service.py:2458]: as all permissions are passed so no need to maintain an array

File: frontend/src/sections/knowledgebase/all-records-page.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  
+  // Navigation helper - updates URL with new params
+  const updateUrl = useCallback((params: Record<string, string | undefined>) => {
+    const newParams = new URLSearchParams();
+    Object.entries(params).forEach(([key, value]) => {
+      if (value !== undefined && value !== null && value !== '') {
+        newParams.set(key, value);
+      }
+    });
+    navigate(`/all-records?${newParams.toString()}`, { replace: true });
```
[the remark this answers is not in the document]
Comment (vishwjeet-pawar03) [reply frontend/src/sections/knowledgebase/all-records-page.tsx]: Fixed

File: frontend/src/sections/knowledgebase/all-records-page.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        sortOrder={sortOrder as 'asc' | 'desc' | undefined}
+        filters={{
+          recordTypes: recordTypes?.split(',').filter(Boolean) || [],
+          origins: origins?.split(',').filter(Boolean) || [],
+          connectorIds: connectorIds?.split(',').filter(Boolean) || [],
+          kbIds: kbIds?.split(',').filter(Boolean) || [],
+          indexingStatus: indexingStatus?.split(',').filter(Boolean) || [],
+          sortBy,
+          sortOrder: sortOrder as 'asc' | 'desc' | undefined,
+        }}
```
[the remark this answers is not in the document]
Comment (vishwjeet-pawar03) [reply frontend/src/sections/knowledgebase/all-records-page.tsx:57]: Fixed

File: frontend/src/sections/knowledgebase/components/all-records-view.tsx [hunk 5]
```diff
@@ -1065,129 +1200,117 @@ const AllRecordsView: React.FC<AllRecordsViewProps> = ({ onNavigateBack, onNavig
...
+          if (node.nodeType === 'record' && node.recordType === 'FILE') {
+            menuActions.push({
+              label: 'Download',
+              icon: downloadIcon,
+              color: theme.palette.primary.main,
+              onClick: () =>
+                handleDownload(
+                  node.origin === ORIGIN.UPLOAD ? node.externalRecordId! : node.id,
+                  node.name
+                ),
```
[the remark this answers is not in the document]
Comment (vishwjeet-pawar03) [reply frontend/src/sections/knowledgebase/components/all-records-view.tsx:1178]: Fixed

**Review by abhishek-pipeshub — APPROVED** [review]
